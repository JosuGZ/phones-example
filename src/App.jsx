import * as React from 'react';
import { connect } from 'react-redux';

import * as Actions from './actions';

import { Loading } from './components/Loading';
import { PhoneListContainer } from './components/PhoneListContainer';

@connect((store) => {
  return {
    phones: store.phones
  };
})
export class App extends React.Component {

  componentDidMount() {
    this.props.dispatch(Actions.getPhones);
  }

  render() {
    return (
      <div className="expand">
        <div className="header">
          <h1>Bienvenido!</h1>
        </div>

        <div className="content">
        {
          (
            this.props.phones !== null ?
              <PhoneListContainer phones={this.props.phones} />
            :
              <Loading />
          )
        }
        </div>
      </div>
    );
  }

}
