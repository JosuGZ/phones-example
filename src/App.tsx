import * as React from 'react';
import { connect } from 'react-redux';

import * as Actions from './actions';

import { Loading } from './components/Loading';
import { PhoneListContainer } from './components/PhoneListContainer';

export interface AppProps {}

interface AppWithDispatchProps extends AppProps {
  dispatch: any;
}

interface AppWithPropsProps extends AppWithDispatchProps {
  phones: any;
}

class AppWithProps extends React.Component<AppWithPropsProps, any> {

  componentDidMount() {
    this.props.dispatch(Actions.getPhones);
  }

  render() {
    console.log('props', this.props);
    return (
      <div className="expand">
        <div className="header">
          <h1>Bienvenido!</h1>
        </div>

        <div className="content">
        {
          (
            this.props.phones !== null ?
              <PhoneListContainer phones={this.props.phones} />
            :
              <Loading />
          )
        }
        </div>
      </div>
    );
  }

}

function mapDispatchToProps(dispatch, props: AppProps): AppWithDispatchProps {
  return {...props, dispatch: dispatch};
}

function mapStateToProps(state: any, props: AppWithDispatchProps): AppWithPropsProps {
  return {...props, phones: state.phones};
}

export const App: React.ComponentClass<AppProps, any> = connect(
  mapStateToProps,
  mapDispatchToProps
) (AppWithProps);
