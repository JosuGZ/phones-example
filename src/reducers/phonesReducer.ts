export const phonesReducer = function(state = null, action) {
  switch (action.type) {
    case 'GET_PHONES':
      return action.payload;
    case 'RESET':
      return null;
    default:
      return state;
  }
};
