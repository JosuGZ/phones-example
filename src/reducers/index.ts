import { combineReducers } from 'redux';

import { phonesReducer } from './phonesReducer';

export const reducer = combineReducers({
  phones: phonesReducer
});
