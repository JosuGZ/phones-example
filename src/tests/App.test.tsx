import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import Thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';

import { reducer } from '../reducers';
import { App } from '../App';

const logger = (store) => (next) => (action) => {
  console.log(action);
  next(action);
};
const middleware = applyMiddleware(Thunk, logger);

const store = createStore(reducer, {
  phones: null
}, middleware);

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Provider store={store}>
        <App />
      </Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
});
