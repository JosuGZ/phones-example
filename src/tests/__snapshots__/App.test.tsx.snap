// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<div>
  <div
    class="expand"
  >
    <div
      class="header"
    >
      <h1>
        Bienvenido!
      </h1>
    </div>
    <div
      class="content"
    >
      <div
        class="loader expand"
        style="text-align: center;"
      >
        <div
          style="position: relative;"
        >
          <svg
            height="30px"
            id="Layer_1"
            version="1.1"
            viewBox="0 0 24 30"
            width="24px"
            x="0px"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
            y="0px"
          >
            <rect
              height="10"
              opacity="0.2"
              width="4"
              x="0"
              y="10"
            >
              <animate
                attributeName="opacity"
                attributeType="XML"
                begin="0s"
                dur="0.6s"
                repeatCount="indefinite"
                values="0.2; 1; .2"
              />
              <animate
                attributeName="height"
                attributeType="XML"
                begin="0s"
                dur="0.6s"
                repeatCount="indefinite"
                values="10; 20; 10"
              />
              <animate
                attributeName="y"
                attributeType="XML"
                begin="0s"
                dur="0.6s"
                repeatCount="indefinite"
                values="10; 5; 10"
              />
            </rect>
            <rect
              height="10"
              opacity="0.2"
              width="4"
              x="8"
              y="10"
            >
              <animate
                attributeName="opacity"
                attributeType="XML"
                begin="0.15s"
                dur="0.6s"
                repeatCount="indefinite"
                values="0.2; 1; .2"
              />
              <animate
                attributeName="height"
                attributeType="XML"
                begin="0.15s"
                dur="0.6s"
                repeatCount="indefinite"
                values="10; 20; 10"
              />
              <animate
                attributeName="y"
                attributeType="XML"
                begin="0.15s"
                dur="0.6s"
                repeatCount="indefinite"
                values="10; 5; 10"
              />
            </rect>
            <rect
              height="10"
              opacity="0.2"
              width="4"
              x="16"
              y="10"
            >
              <animate
                attributeName="opacity"
                attributeType="XML"
                begin="0.3s"
                dur="0.6s"
                repeatCount="indefinite"
                values="0.2; 1; .2"
              />
              <animate
                attributeName="height"
                attributeType="XML"
                begin="0.3s"
                dur="0.6s"
                repeatCount="indefinite"
                values="10; 20; 10"
              />
              <animate
                attributeName="y"
                attributeType="XML"
                begin="0.3s"
                dur="0.6s"
                repeatCount="indefinite"
                values="10; 5; 10"
              />
            </rect>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
